Sources/BraintreeApplePay/BTApplePayClient.swift

```diff
@@ -115,11 +110,25 @@ import BraintreeCore
         ]
     }
 
-    func completionHandler(
-        onMainThreadWithPaymentRequest paymentRequest: PKPaymentRequest?,
-        error: Error?,
+    // MARK: - Analytics Completion Helpers
+
+    private func completionWithAnalytics(
+        for analyticsResult: BTAnalyticsResult,
+        nonce: BTApplePayCardNonce? = nil,
+        error: Error? = nil,
+        completion: @escaping (BTApplePayCardNonce?, Error?) -> Void
+    ) {
```

Comment:
I think these helper methods are a bit over-loaded, especially since we expect either only an error or a nonce, but never both.

Maybe a good compromise is reverting back to using two separate methods for calling success vs fail cases, but then also performing the concat using the associated string value from an enum.

Reply:
That sounds good to me. Are we tied to the names `notifySuccess/notifyFailure` or can we do something like `failureCompletionWithAnalytics/successCompletionWithAnalytics`? Reason being that notify still feels tied to something like a delegate to me. I’m fine either way, just wanted to give a why. 🙂

For the enum are you thinking doing `.stringValue` or actually passing the whole string in like `.success(apple-pay:tokenize:succeeded)`. Just want to make sure we are on the same page since we’ve talked about both here!

I can make the updates tomorrow in any case.

Reply:
Implemented what we had discussed here: https://github.com/braintree/braintree_ios/pull/960#discussion_r1122402533

If we like this pattern, we can make this update in the other 2 PRs, but we didn't want to make the change in all 3 places since the discussion was still ongoing. This PR now reflects the changes discussed above and added a test file for our event names.